model/qti/container/Container.php

```diff
@@ -188,6 +190,29 @@ public function checkIntegrity($body, &$missingElements = null){
     }
 
     /**
+     * Converts <foo /> to <foo></foo> unless foo is a proper void element such as img etc.
+     *
+     * @param $html
+     * @return mixed
+     */
+    public function fixNonvoidTags($html) {
+        return preg_replace_callback('~(<([\w]+)[^>]*?)(\s*/>)~', function ($matches) {
+            // something went wrong
+            if(empty($matches[2])) {
+                // do nothing
+                return $matches[0];
+            }
+            // regular void elements
+            if(in_array($matches[2], array('area','base','br','col','embed','hr','img','input','keygen','link','meta','param','source','track','wbr'))) {
+                // do nothing
+                return $matches[0];
+            }
+            // correctly closed element
+            return trim(substr($matches[0], 0, -2)) . '></' . $matches[2] . '>';
```

Comment:
PHP's substr is not binary-safe. Please use mb_substr with 'UTF-8' encoding to make sure we do not break funny characters.